framework/include/materials/MaterialData.h

```diff
@@ -155,8 +158,13 @@ class MaterialData
   ///@}
 
   /**
-   * Resizes the properties to the specified size
-   * @param size The size of the properties to set
+   * Resizes the number of properties to the specified size (including
+   * stateful old and older properties.
+   *
+   * Warning:
+   * If size is greater than the current size, the new properties are set
+   * to NULL except * for the last element - which is set to a blank material
```

Comment:
I've fixed in in updates to the PR.  Relevant section is here: https://github.com/idaholab/moose/pull/8278/files#diff-a7d1c85c03658594a218c1dc277310efL217